### DISCUSSION TITLE
Cluster installation issue

### DISCUSSION THREAD

**Discussion by shweta-shinde** [okd-project/okd#1105] [Q&A]
hello,
I am installing an openshift cluster in vsphere using upi.
i am able to bring up the bootstrap and master node.
But the working nodes are failing to boot up, Here is there error i am getting on the worker vm
GET https://api-int.test.ocpilab.com:22623/config/worker
GET result: Internal Server error

I am not able to ssh into the master node with the ssh key configured in install-config.yaml file.

I checked the loadbalancer logs 
2022/02/09 06:26:50 [error] 1878#1878: *237538 connect() failed (111: Connection refused) while connecting to upstream, client: 161.92.211.232, server: 0.0.0.0:22623, upstream: "161.92.211.236:22623", bytes from/to client:0/0, bytes from/to upstream:0/0
The connection from worker to master is being refused.

**Comment by vrutkovs** [comment]
>working nodes are failing to boot up

We disallow sending /config/worker until bootstrap is up to prevent configuration drift. These machines will boot when machine-config-server would run on master nodes

**Reply by shweta-shinde** [reply]
so the issue is with the bootstrap node, is it not configured properly?
Because I can see the login prompt after booting in bootstrap and master nodes.

**Reply by vrutkovs** [reply]
>is it not configured properly?

It works as designed - you need to take down the bootstrap node after master nodes are running MCS

**Reply by shweta-shinde** [reply]
yes, even tried that , I took down the bootstrap node after configuring the master nodes.
Got connection refused from https://api-int.test.ocpilab.com:22623/config/worker

**Reply by shweta-shinde** [reply]
i tried curl -x "" -v https://api-int.test.ocpilab.com:22623/config/worker --insecure -->
* About to connect() to api-int.test.ocpilab.com port 22623 (#0)
*   Trying 161.92.211.232...
* Connected to api-int.test.ocpilab.com (161.92.211.232) port 22623 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* NSS error -5961 (PR_CONNECT_RESET_ERROR)
* TCP connection reset by peer
* Closing connection 0
curl: (35) TCP connection reset by peer

**Comment by arlozano** [comment]
Hi shweta-shinde
In case you still won't be able to install, I had the same issue, after a week discover why, the fcos ova version was incorrect, in the documentation does not appear but it MUST be a specific ova version. You can find which version is needed with this command:
$ openshift-install coreos print-stream-json | grep vmware
                "vmware": {
                                "location": "https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/34.20210626.3.1/x86_64/fedora-coreos-34.20210626.3.1-vmware.x86_64.ova",
                                "signature": "https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/34.20210626.3.1/x86_64/fedora-coreos-34.20210626.3.1-vmware.x86_64.ova.sig",

Hope it helps.

**Reply by shweta-shinde** [reply]
Hi @arlozano 
I used the the version from the command you suggested.
Still the worker node is not booting up
this this the log i am seeing from worker console
GET error: Get "https://api-int.test.ocpilab.com:22623/config/worker": EOF

**Reply by arlozano** [reply]
Hi @shweta-shinde 
The url will be up when the masters node get online, not from the bootstrap. Does "https://api-int.test.ocpilab.com:22623/config/master" work ? can you get the masters node states with oc get nodes?
If you can get the masters state then you will have to approve the sign cert requests to complete masters installation:
oc get csr | grep -i pending
for each pending request:
oc adm certificate approve csr-fxj8z <-- the pending request obtained before
After that your masters nodes will be online and the worker config url available.
